### PR TITLE
feat: mandatory email reminder for attendees with @gmail.com

### LINKED ISSUE #10247: [CAL-2204] RFC: mandatory email reminder for attendees with @gmail.com
due to googles new privacy policy, events are not automatically added for @gmail.com emails. They first need to opt-in "I know this person" before getting email invites.

I bet there are people missing events because of this.

We should send a mandatory email reminder to attendees with @gmail.com to be extra safe, maybe 15 min before the meeting? Or one hour? Or both?

<sub>[CAL-2204](https://linear.app/calcom/issue/CAL-2204/rfc-mandatory-email-reminder-for-attendees-with-gmailcom)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Changes from: https://github.com/calcom/cal.com/pull/12647

By: [Chiranjeev-droid](https://github.com/Chiranjeev-droid)

Fixes https://github.com/calcom/cal.com/issues/10247

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (Unit/Integration/E2E or any other test)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by CarinaWolli** [comment]
@Udit-takkar could you review this for me? 🙏 Changes are all coming from https://github.com/calcom/cal.com/pull/12647 (expect the latest commit with some small fixes [Link to Commt](https://github.com/calcom/cal.com/pull/12747/commits/5b3f558993e950645f82a12c4c5c813498da028f)) 

I already reviewed and it was good to go from my side

File: packages/features/ee/workflows/lib/reminders/emailReminderManager.ts [hunk 1]
```diff
@@ -105,12 +105,13 @@ export const scheduleEmailReminder = async (
...
   emailBody: string,
-  workflowStepId: number,
   template: WorkflowTemplates,
   sender: string,
+  workflowStepId?: number,
   hideBranding?: boolean,
   seatReferenceUid?: string,
-  includeCalendarEvent?: boolean
+  includeCalendarEvent?: boolean,
+  isMandatoryReminder?: boolean
```
Comment (Udit-takkar) [inline packages/features/ee/workflows/lib/reminders/emailReminderManager.ts:114]: we should use object as parameter as they are too many parameters and some are even optional

File: packages/features/ee/workflows/lib/reminders/emailReminderManager.ts [hunk 2]
```diff
@@ -311,32 +312,59 @@ export const scheduleEmailReminder = async (
...
+              scheduledDate: scheduledDate.toDate(),
+              scheduled: true,
+              referenceId: batchId,
+              seatReferenceId: seatReferenceUid,
+              isMandatoryReminder: true,
+            },
+          });
+        }
+      } catch (error) {
+        console.log(`Error scheduling email with error ${error}`);
```
Comment (Udit-takkar) [inline packages/features/ee/workflows/lib/reminders/emailReminderManager.ts:341]: NIT: use log.error

**Review by Udit-takkar — DISMISSED** [review]
code LGTM. left some comments

File: packages/features/ee/workflows/lib/reminders/emailReminderManager.ts (hunk 1 above)
Comment (CarinaWolli) [reply packages/features/ee/workflows/lib/reminders/emailReminderManager.ts]: Agree, changed to object parameter for `scheduleEmailReminder`, `scheduleSMSReminder` and `ScheduleWhatsappReminder`

**Review by PeerRich — APPROVED** [review]
approved on behalf of @Udit-takkar

**Pull request merged**